Core/src/org/sleuthkit/autopsy/texttranslation/translators/BingTranslatorSettingsPanel.java

```diff
@@ -0,0 +1,316 @@
+            targetLanguageCode = selectedCode;
+            firePropertyChange("SettingChanged", true, false);
+        }
+    }//GEN-LAST:event_targetLanguageComboBoxSelected
+
+    // Variables declaration - do not modify//GEN-BEGIN:variables
+    private javax.swing.JTextField authenticationKeyField;
+    private javax.swing.JLabel authenticationKeyLabel;
+    private javax.swing.JLabel resultLabel;
+    private javax.swing.JComboBox<LanguageWrapper> targetLanguageComboBox;
+    private javax.swing.JLabel targetLanguageLabel;
+    private javax.swing.JButton testButton;
+    private javax.swing.JLabel testResultValueLabel;
+    private javax.swing.JTextField testUntranslatedTextField;
+    private javax.swing.JLabel untranslatedLabel;
+    private javax.swing.JLabel warningLabel;
+    // End of variables declaration//GEN-END:variables
+    /**
+     * Converts an input test to the JSON format required by Bing Translator,
+     * posts it to Microsoft, and returns the JSON text response.
+     *
+     * @param string The input text to be translated.
+     *
+     * @return The translation response as a JSON string
+     *
+     * @throws IOException if the request could not be executed due to
+     *                     cancellation, a connectivity problem or timeout.
+     */
+    private boolean testTranslationSetup() {
+        testResultValueLabel.setText("");
+        MediaType mediaType = MediaType.parse("application/json");
+        JsonArray jsonArray = new JsonArray();
+        JsonObject jsonObject = new JsonObject();
+        jsonObject.addProperty("Text", testUntranslatedTextField.getText());
+        jsonArray.add(jsonObject);
+        String bodyString = jsonArray.toString();
+
+        RequestBody body = RequestBody.create(mediaType,
+                bodyString);
+        Request request = new Request.Builder()
+                .url(BingTranslator.getTranlatorUrl(targetLanguageCode)).post(body)
+                .addHeader("Ocp-Apim-Subscription-Key", authenticationKeyField.getText())
+                .addHeader("Content-type", "application/json").build();
+        try {
+            Response response = new OkHttpClient().newCall(request).execute();
+            JsonParser parser = new JsonParser();
+            JsonArray responses = parser.parse(response.body().string()).getAsJsonArray();
+            //As far as I know, there's always exactly one item in the array.
+            JsonObject response0 = responses.get(0).getAsJsonObject();
+            JsonArray translations = response0.getAsJsonArray("translations");
+            JsonObject translation0 = translations.get(0).getAsJsonObject();
+            testResultValueLabel.setText(translation0.get("text").getAsString());
+            return true;
+        } catch (IOException | IllegalStateException | ClassCastException | NullPointerException | IndexOutOfBoundsException e) {
+            return false;
```

Comment:
Doesn't this exception need to be logged?

Reply:
Yeah that certainly seems like a good call here.